test/support/src/helpers.cc

```diff
@@ -53,6 +55,14 @@ std::mutex catch2_macro_mutex;
 namespace tiledb {
 namespace test {
 
+const std::string& get_temp_path() {
+  // Ensure the path has a trailing delimiter.
+  static std::string temp_path =
+      (std::filesystem::temp_directory_path() / "tiledb_test" / "").string();
```

Comment:
I had tried adding the PID to the path to avoid collisions, but the soonest I added an `#include <windows.h>` I started getting weird compiler errors.

Reply:
Maybe try adding your own extern definition for getpid() rather than including the windows header?

Reply:
We do have internally generated UUIDs (currently in `tiledb/sm/misc/uuid.h`) if you want something that is not platform dependent but can still be used to avoid collisions.